subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -163,6 +163,192 @@ class ArtifactTransformIntegrationTest extends AbstractIntegrationSpec {
         file("app/build/libs").listFiles().size() == 1
     }
 
+    def "User gets a reasonable error message when a transformation throws exception"() {
+        given:
+        buildFile << """
+            import org.gradle.api.artifacts.transform.*
+
+            apply plugin: 'java'
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                compile 'com.google.guava:guava:19.0'
```

Comment:
In general, we should avoid tests that reach out to maven central, and instead use the http or local file repository test fixtures on `AbstractDependencyResolutionTest` and `AbstractHttpDependencyResolutionTest`. These superclasses also take care of isolating the dependency resolution tests from each other.